scripts/install/install-helper-functions.sh

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash 
+# This file is to be sourced into another to add helper functions, rather than
+# executed directly.
+
+#-------------------------------------------------------------------------------
+# Logging support (nascent; I want to add more control and a log file).
+
+# Print a separator bar of # characters.
+function echo_bar() {
+  local terminal_width
+  if [ -n "${TERM}" ] && [ -t 0 ] ; then
+    if [[ -n "$(which resize)" ]] ; then
+      terminal_width="$(resize 2>/dev/null | grep COLUMNS= | cut -d= -f2)"
+    elif [[ -n "$(which stty)" ]] ; then
+      terminal_width="$(stty size 2>/dev/null | cut '-d ' -f2)"
+    fi
+  fi
+  printf "%${terminal_width:-80}s\n" | tr ' ' '#'
+}
+
+function echo_running_sudo() {
+  if [ "$(id -u -n)" == "root" ] ; then
+    echo "Running $1"
+  else
+    echo "Running sudo $1; you may be prompted for your password."
+  fi
+}
+
+#-------------------------------------------------------------------------------
+# Misc helper functions.
+
+# Execute the args (a command) as root, either using sudo or directly if already
+# running as root.
+function my_sudo() {
+  if [ "$(id -u -n)" == "root" ] ; then
+    "$@"
+  else
+    (set -x ; sudo "$@")
+  fi
+}
+
+# Get the type of the first arg, i.e. shell function, executable, etc.
+# For more info: https://ss64.com/bash/type.html
+#            or: https://bash.cyberciti.biz/guide/Type_command
+function safe_type() {
+  type -t "${1}" || /bin/true
+}
+
+# Print the disk location of the first arg.
+function safe_which() {
+  type -p "${1}" || /bin/true
+}
+
+# Does the first arg start with the second arg?
+function beginswith() { case "${1}" in "${2}"*) true;; *) false;; esac; }
+
+#-------------------------------------------------------------------------------
+# Helpers for finding the APT caching proxy.
+
+# 
+function proc_net_route_addr() {
```

Comment:
There's no way to query the docker command for this info?

Reply:
Not that I know of. I think it is different for each container, and the docker command is not available inside the container. My research about the problem let to code very much like this.